baseapp/baseapp.go

```diff
@@ -761,3 +768,37 @@ func (app *BaseApp) runMsgs(ctx sdk.Context, msgs []sdk.Msg, mode runTxMode) (*s
 func makeABCIData(msgResponses []*codectypes.Any) ([]byte, error) {
 	return proto.Marshal(&sdk.TxMsgData{MsgResponses: msgResponses})
 }
+
+func generateFraudProof(store *multi.Store, storeKeyToSubstoreTraceBuf map[types.StoreKey]*bytes.Buffer) FraudProof {
+	var fraudProof FraudProof
+	fraudProof.stateWitness = make(map[string]StateWitness)
+
+	for storeKey, subStoreTraceBuf := range storeKeyToSubstoreTraceBuf {
+		kvStore := store.GetKVStore(storeKey)
+		traceKv := kvStore.(*tracekv.Store)
+		keys := traceKv.GetAllKeysUsedInTrace(*subStoreTraceBuf)
+
+		substoreSMT := store.GetSubStoreSMT(storeKey.Name())
+
+		root := substoreSMT.Root()
+
+		var stateWitness StateWitness
+		stateWitness.root = root
+
+		for key := range keys {
+			var witnessData WitnessData
+			value := substoreSMT.Get([]byte(key))
+			proof, err := substoreSMT.GetSMTProof([]byte(key))
+			if err != nil {
+				panic(err)
+			}
```

Comment:
Usually, this is wrapped via a method prefixed with `Must*`, e.g. here it would be `MustGetSMTProof`. Though I sometimes prefer that a panic is commented (why is it actually OK to panic in this case? Why is this either supposed to never happen or why can't we recover from it and moust immediately stop everything?)

Reply:
☝️ @Manav-Aggarwal

Reply:
Added a `MustGetSMTProof` and commented in why it is okay to panic.